### ISSUE TITLE
Stub generator

### ISSUE THREAD

**Issue by timwoocker** [py4j/py4j#514]
The biggest problem I'm having with using py4j is that there is no autocompletion for the java classes.
It would be a great addition to have a tool that converts a java project into stub files for py4j. It could be created in a separate repository as it would be a standalone tool.

What do you guys think about that? A tool like ANTLR could be used for the syntax parsing, creating python stub files out of that shouldn't be too hard as the implementation isn't relevant and the generated stub project structure could be pretty much the same as the java project.

An alternative approach would be to write a plugin for IDEs (PyCharm or VS Code) that provides autocompletion based on .java source files for classes accessed through the gateway.

**Comment by HyukjinKwon** [comment]
I don't think we can do a static analysis there because we can only access to the Java instances when the server is up (runtime).

**Comment by HyukjinKwon** [comment]
in PySpark case (that uses Py4J), we just manually type all cases that we expose as an API.

**Comment by HyukjinKwon** [comment]
Other notebooks like Jupyter IIRC support the automcompletion by doing `dir(...)` under the hood so that might be one example that we can follow.
